enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/readreplica/ReadReplicaStartupProcess.java

```diff
@@ -41,18 +42,23 @@
 
 class ReadReplicaStartupProcess implements Lifecycle
 {
+    private static final int MAX_ATTEMPTS = 5;
```

Comment:
To be honest I think this should just be redesigned, so I would rather not consider that point for this PR. If this was exposed in settings it should rather be something like a maximum time to keep trying before giving up.